File: frontend/src/domain/entities/Technology.ts
export class Technology {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly slug: string,
    public readonly iconUrl: string | null,
    public readonly category: string | null,
    public readonly createdAt: Date,
    public readonly projectCount: number = 0
  ) {}

  hasProjects(): boolean {
    return this.projectCount > 0;
  }
}

File: backend/api/schemas/technology.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, Field


class TechnologyBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    slug: str = Field(..., min_length=1, max_length=100)
    icon_url: Optional[str] = None
    category: Optional[str] = None


class TechnologySchema(TechnologyBase):
    id: UUID
    created_at: datetime
    project_count: int = Field(default=0)

    class Config:
        from_attributes = True


class TechnologyCreateSchema(TechnologyBase):
    pass
